<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Input Files &mdash; PangyPlot 0.1.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
    <link rel="shortcut icon" href="_static/favicon.svg"/>
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Choosing Parameters" href="parameters.html" />
    <link rel="prev" title="Local Build" href="local_build.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            PangyPlot
              <img src="_static/pangyplot_logo.svg" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                0.1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="local_build.html">Local Build</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Input Files</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#quick-summary">Quick Summary</a></li>
<li class="toctree-l2"><a class="reference internal" href="#multi-sample-vcf">Multi-sample VCF</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#variant-specific-columns">Variant Specific Columns</a></li>
<li class="toctree-l3"><a class="reference internal" href="#sample-specific-columns">Sample Specific Columns</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#sample-information-file">Sample Information File</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#sample-id">Sample ID</a></li>
<li class="toctree-l3"><a class="reference internal" href="#phenotype">Phenotype</a></li>
<li class="toctree-l3"><a class="reference internal" href="#group-id">Group ID</a></li>
<li class="toctree-l3"><a class="reference internal" href="#read-depth">Read Depth</a></li>
<li class="toctree-l3"><a class="reference internal" href="#covariates">Covariates</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#bed-file-optional">BED File (Optional)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="parameters.html">Choosing Parameters</a></li>
<li class="toctree-l1"><a class="reference internal" href="tests.html">Association Tests</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="power_calculation.html">Power Simulation Package</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PangyPlot</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Input Files</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/input.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="input-files">
<span id="input"></span><h1>Input Files<a class="headerlink" href="#input-files" title="Permalink to this heading"></a></h1>
<section id="quick-summary">
<h2>Quick Summary<a class="headerlink" href="#quick-summary" title="Permalink to this heading"></a></h2>
<p>VikNGS takes 3 different file types as input:</p>
<ul class="simple">
<li><dl class="simple">
<dt>a multi=sample VCF that provides genotype information</dt><dd><ul>
<li><p>genotype information is extracted from <strong>GL</strong> (preferred), <strong>PL</strong>, and <strong>GT</strong> fields</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>a tab-separated sample information text file containing phenotype and covariate information</dt><dd><ul>
<li><p>must be created by user ref:<cite>see column details code&lt;sample_info&gt;</cite> for what this file should contain</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>a BED file to specify the variant collapsing strategy (optional)</dt><dd><ul>
<li><p>genes and exons specified in the file can be used to define regions to collapse upon</p></li>
<li><p>can be generated automatically from the <a class="reference external" href="http://genome.ucsc.edu/cgi-bin/hgTables">UCSC Table Browser</a></p></li>
</ul>
</dd>
</dl>
</li>
</ul>
<p>Below, the different types of files are explained in further detail.-</p>
</section>
<section id="multi-sample-vcf">
<span id="multisample-vcf"></span><h2>Multi-sample VCF<a class="headerlink" href="#multi-sample-vcf" title="Permalink to this heading"></a></h2>
<p>A Variant Call Format (VCF) file is a standard way of storing variant information called from sequencing data. Each row of a VCF file corresponds to a genetic variant (insertion, deletion or substitution) and contains information such as the genomic position of the variant, the confidence in the variant call and many other additional annotations.</p>
<p>A multi-sample VCF is formatted identically to a single-sample VCF except it contains an extra set of columns corresponding to sample-specific data.</p>
<figure class="align-center" id="id1">
<img alt="Multsample VCF Layout" src="_images/vcf_layout.png" />
<figcaption>
<p><span class="caption-text">The general layout of a multi-sample VCF file.</span><a class="headerlink" href="#id1" title="Permalink to this image"></a></p>
</figcaption>
</figure>
<p>The first set of lines in a VCF file make up the header and are denoted by the characters <em>##</em>. The header includes information about the data source and how the VCF file was constructed. This information is ignored when by VikNGS. The last line of the header is denoted with a single <em>#</em> and includes the column names in addition to a unique identifier for every sample. The first nine columns must be (tab-delimited, in order) <strong>CHROM</strong>, <strong>POS</strong>, <strong>ID</strong>, <strong>REF</strong>, <strong>ALT</strong>, <strong>QUAL</strong>, <strong>FILTER</strong>, <strong>INFO</strong> and <strong>FORMAT</strong>. Every subsequent value is expected to be a unique sample identifier.</p>
<div class="literal-block-wrapper docutils container" id="id2">
<div class="code-block-caption"><span class="caption-text"><em>Example of a multi-sample VCF</em></span><a class="headerlink" href="#id2" title="Permalink to this code"></a></div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span> <span class="c1">##fileformat=VCFv4.1</span>
 <span class="c1">##FILTER=&lt;ID=PASS,Description=&quot;All filters passed&quot;&gt;</span>
 <span class="c1">##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;</span>
 <span class="c1">##FORMAT=&lt;ID=GL,Number=.,Type=Integer,Description=&quot;Genotype Likelihood&quot;&gt;</span>
 <span class="c1">##FORMAT=&lt;ID=PL,Number=G,Type=Integer,Description=&quot;Normalized, Phred-scaled likelihoods&quot;&gt;</span>
 <span class="c1">#CHROM      POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  SAMPLE_1_ID     SAMPLE_2_ID     ...</span>
   <span class="mi">22</span>        <span class="mi">160036</span>  <span class="n">snp_1</span>   <span class="n">A</span>       <span class="n">C</span>       <span class="o">.</span>       <span class="n">PASS</span>    <span class="o">.</span>       <span class="n">GT</span><span class="p">:</span><span class="n">PL</span><span class="p">:</span><span class="n">GL</span>        <span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">15</span><span class="p">,</span><span class="mi">73</span><span class="p">:</span><span class="o">.</span>   <span class="mi">1</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">50</span><span class="p">,</span><span class="mi">8</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="o">-</span><span class="mi">5</span><span class="p">,</span><span class="o">-</span><span class="mf">0.84</span><span class="p">,</span><span class="o">-</span><span class="mf">0.07</span>
   <span class="mi">22</span>        <span class="mi">160408</span>  <span class="n">snp_2</span>   <span class="n">T</span>       <span class="n">C</span>       <span class="o">.</span>       <span class="n">PASS</span>    <span class="o">.</span>       <span class="n">GT</span><span class="p">:</span><span class="n">PL</span>   <span class="o">./.</span><span class="p">:</span><span class="o">.</span>   <span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">36</span><span class="p">,</span><span class="mi">248</span>
   <span class="mi">22</span>        <span class="mi">160612</span>  <span class="n">snp_2</span>   <span class="n">C</span>       <span class="n">G</span>       <span class="o">.</span>       <span class="n">PASS</span>    <span class="o">.</span>       <span class="n">GT</span><span class="p">:</span><span class="n">GL</span>   <span class="mi">0</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mf">0.48</span><span class="p">,</span><span class="o">-</span><span class="mf">0.48</span><span class="p">,</span><span class="o">-</span><span class="mf">0.48</span>   <span class="mi">1</span><span class="o">/</span><span class="mi">1</span><span class="p">::</span><span class="o">-</span><span class="mf">4.4</span><span class="p">,</span><span class="o">-</span><span class="mf">0.27</span><span class="p">,</span><span class="o">-</span><span class="mf">0.33</span>
</pre></div>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>While parsing the file only SNPs will be retained (a single A, T, C, or G in both the REF and ALT columns). Multiallelic sites are also currently ignored by VikNGS.</p>
</div>
<section id="variant-specific-columns">
<h3>Variant Specific Columns<a class="headerlink" href="#variant-specific-columns" title="Permalink to this heading"></a></h3>
<p>The first nine columns contain information relevant to the called variants.</p>
<p>The first two required columns (<strong>CHROM</strong>, <strong>POS</strong>) specify the genomic coordinates of the variant. The first base pair of a chromosome is denoted as position 1. <strong>CHROM</strong> is treated as a unique identifier for a chromosome. <strong>POS</strong> is expected to be a positive integer value.</p>
<p>The third column (<strong>ID</strong>) assigns an identifier to each variant.</p>
<p>The forth and fifth columns (<strong>REF</strong>, <strong>ALT</strong>) refer to reference alleles derived from a reference genome and alternative alleles observed in the sequencing data. These columns specify whether the variant is an indel or a SNP.</p>
<p>The sixth column (<strong>QUAL</strong>) provides a Phred-scaled probability that the called base is correct. This field is ignored and not used by vikNGS.</p>
<p>The seventh column (<strong>FILTER</strong>) Indicates whether or not the variant has passed all of the quality control checks present in the variant calling pipeline that produced the VCF file. The value of this field will be <strong>PASS</strong> if the variant passed all of the filters and will be “.” if no checks were applied.</p>
<p>The eighth column (<strong>INFO</strong>) contains additional variant annotations, delimited by a semicolon. This field is ignored and not used by vikNGS.</p>
<p>The ninth column (<strong>FORMAT</strong>) specifies the ordering of sample-level annotations, delimited by a colon. All subsequent sample-specific columns will provide values for each annotation, placed in the same order and also delimited by a colon. In particular, the ordering of <strong>GT</strong>, <strong>GL</strong>, and <strong>PL</strong> is extracted from this column.</p>
</section>
<section id="sample-specific-columns">
<h3>Sample Specific Columns<a class="headerlink" href="#sample-specific-columns" title="Permalink to this heading"></a></h3>
<p>All columns following the <strong>FORMAT</strong> column should correspond to a single sample. The header of each column specifies a unique identifier for each sample. Values in subsequent columns are colon separated and are ordered with respect to the labels in the <strong>FORMAT</strong> column. The following values are extracted for each sample.</p>
<p><strong>GT</strong> (Genotype): A pair of values specifying the predicted genotype for a specific sample (0/0 homozygous reference, 0/1 heterozygous, 1/1 homozygous alternative).</p>
<p><strong>GL</strong> (Genotype likelihood): A set of three numbers indicating the log-scaled genotype probability of all three possible genotypes. Each number corresponds to the value for homozygous reference, heterozygous, and homozygous alternative, respectively.</p>
<p><strong>PL</strong> (Phred-scaled likelihood): A set of three integers indicating the Phred-scaled probability of all three possible genotypes (homozygous reference, heterozygous, and homozygous alternative). These values are normalized since all the values are relative to the most likely genotype (and are therefore not probabilities).</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>GL</strong> and <strong>PL</strong> are both ways of representing genotype probabilities. Both values can be used to derive genotype probabilities but because <strong>PL</strong> is a rounded integer, <strong>GL</strong> is more accurate.</p>
<p><span class="math notranslate nohighlight">\(genotype\_probability=10^{\textbf{GL}}=10^{-\textbf{PL}/10}\)</span></p>
<p><span class="math notranslate nohighlight">\(-10\textbf{GL}=\textbf{PL}\)</span></p>
</div>
<p>When running an association test in VikNGS, different genotype values can potentially be used. If the “Use VCF GT” option is checked, VikNGS will extract the <strong>GT</strong> value only and convert the value to a genotype {0, 1, 2}. If “Use Expected GT” is checked, the software will attempt to parse the <strong>GL</strong> values from each sample first. If <strong>GL</strong> values are missing or are formatted incorrectly, then the <strong>PL</strong> values will be extracted. If parsing of both <strong>GL</strong> and <strong>PL</strong> values fail, the <strong>GT</strong> values will be extracted. An expected genotype value [0,2] will be calculated from the extracted column. If “Use GT calls” is chosen, genotype probabilities will be extracted in the same way the expected gentype method but a hard genotype call {0, 1, 2} will be made instead of calculating the expected value.</p>
<blockquote>
<div><p>If the relevant information is not present for a given variant, that variant will be skipped and not included in analysis.</p>
</div></blockquote>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Note that if <strong>GT</strong> values are indicated as “missing” (ex. ./.) then the variant will be skipped even if values for <strong>GL</strong> and <strong>PL</strong> are present.</p>
</div>
</section>
</section>
<section id="sample-information-file">
<span id="sample-info"></span><h2>Sample Information File<a class="headerlink" href="#sample-information-file" title="Permalink to this heading"></a></h2>
<p>To utilize phenotypic data and sample-specific information, vikNGS requires the user to provide this information in a separate file. This tab-separated file is defined specifically for use in vikNGS. This file should <em>not</em> have headers and is expected to contain one sample per line.</p>
<p>The columns are defined as follows:</p>
<section id="sample-id">
<h3>Sample ID<a class="headerlink" href="#sample-id" title="Permalink to this heading"></a></h3>
<p>Every line in the sample information file should begin with a <em>unique</em> sample ID. The only additional requirement is that every sample ID needs to identically match exactly one of the IDs that appear after the <strong>FORMAT</strong> column in the multisample VCF file. This column specifies the relationship between the sample-specific data and the data in the VCF file.</p>
</section>
<section id="phenotype">
<h3>Phenotype<a class="headerlink" href="#phenotype" title="Permalink to this heading"></a></h3>
<p>This column contains phenotypic data which will be used to identify association with genotype information.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If looking to find association information between case-control groups, this column is used to specify case-control status. Please designate cases with a 1 and controls with a 0 in this column.</p>
</div>
</section>
<section id="group-id">
<h3>Group ID<a class="headerlink" href="#group-id" title="Permalink to this heading"></a></h3>
<p>Use this column to specify if samples are from different groups or studies. Any samples with the same value in the column will be put in the same group.</p>
</section>
<section id="read-depth">
<h3>Read Depth<a class="headerlink" href="#read-depth" title="Permalink to this heading"></a></h3>
<p>If using the expected genotype method, the score test calculates the variance for each group separately and needs to be aware of which groups are high read depth versus low read depth. The read depth of each sample must be specified in this column. The numerical read depth value (ex. 32) can be provided for each sample or simply a letter specifying whether a sample is from a high or low sequencing run (H = high, L = low). Note that all samples with a shared group ID must also share high/low read depth status. Therefore, the first read depth value encountered for a group will be applied to all members of that group.</p>
</section>
<section id="covariates">
<h3>Covariates<a class="headerlink" href="#covariates" title="Permalink to this heading"></a></h3>
<p>The remaining columns are used to specify covariates. Covariates can either be continuous or categorical. If every value in a covariate column is numeric, the column will be treated as a continuous covariate. If a single non-numeric value is identified, the covariate will be treated as categorical and a new dummy covariate will be made for EVERY unique value (high cardinality categorical variables can result in significantly longer computation time).</p>
</section>
</section>
<section id="bed-file-optional">
<span id="bed-file"></span><h2>BED File (Optional)<a class="headerlink" href="#bed-file-optional" title="Permalink to this heading"></a></h2>
<p>Elucidation of associated rare variants can be challenging because the frequency of the associated allele can be extremely low. To improve the power of statistical tests that identify rare alleles, it is necessary to collapse a group of linked variants and perform the association test on a genetic region rather than individual SNPs. For rare variant association in vikiNGS, a collapsing strategy must be specified.</p>
<p>There are three types of collapsing strategies available:
- Collapse every \(k\)
- Collapse by gene
- Collapse by exon</p>
<p>By default, the variants will be read and filtered from an input VCF file. After the filtering step, the first \(k\)th variants will be collapsed together, followed by the next set of \(k\) non-overlapping variants and so on (\(k=5\) by default).</p>
<p>To collapse variants in a more biologically relevant way, a BED file must be provided specifying the collapsible regions. A BED file is a tab-delimited table which describes genomic features intended to be used for visualization in a genome browser. The format is specified by UCSC Genome Bioinformatics, detailed information can be found on their <a class="reference external" href="https://genome.ucsc.edu/FAQ/FAQformat.html">web page</a> web page. Every line describes a single region as follows</p>
<p>The first three columns specify the gene</p>
<ol class="arabic simple">
<li><p><strong>chrom</strong> - The name of the chromosome matching the first column in the VCF file.</p></li>
<li><p><strong>chromStart</strong> - The starting position of the gene on the chromosome (starting from 0)</p></li>
<li><p><strong>chromEnd</strong> - The ending position of the gene on the chromosome. This base is not included in the gene.</p></li>
</ol>
<p>For example, to specify the first 250 bases on chromosome 4:
chr4    0    250</p>
<p>The next six column are specified by the BED format but are not used in variant collapsing:</p>
<ol class="arabic simple" start="4">
<li><p><strong>name</strong> - Optional identifier for this region.</p></li>
<li><p><strong>score</strong> -  Not used.</p></li>
<li><p><strong>strand</strong> - Not used.</p></li>
<li><p><strong>thickStart</strong> - Not used.</p></li>
<li><p><strong>thickEnd</strong> - Not used.</p></li>
<li><p><strong>itemRgb</strong> - Not used.</p></li>
</ol>
<p>The last three columns are potentially used if collapsing:</p>
<ol class="arabic simple" start="10">
<li><p><strong>blockCount</strong> - The number of blocks (exons) in the gene.</p></li>
<li><p><strong>blockSizes</strong> - The size of each exon, comma separated list the size of blockCount.</p></li>
<li><p><strong>blockStarts</strong> - Positions where each exon should begin, relative to chromStart. Comma separated list the size of blockCount</p></li>
</ol>
<p>To collapse variants by gene, the first three columns are required to indicate where each gene begins and ends.</p>
<p>To collapse variants by exon, all twelve columns must be present. The coding region is defined to be where the first block/exon starts to where the last one ends. Each exon is specified by a block and variants within that block will be collapsed.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Genes and exons can be very large and could contain thousands of variants to collapse in a single test. This can cause large computational burden, especially if a permutation test is used to calculate the p-value. Therefore, a maximum collapse size can be specified. Variants will be collapsed into a single test until the maximum size is reached and subsequent variants will be put into a new collapsed set.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="local_build.html" class="btn btn-neutral float-left" title="Local Build" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="parameters.html" class="btn btn-neutral float-right" title="Choosing Parameters" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Scott Mastromatteo.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>