<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Choosing Parameters &mdash; PangyPlot 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css" />
      <link rel="stylesheet" type="text/css" href="_static/pangyplot.css" />

  
    <link rel="shortcut icon" href="_static/favicon.svg"/>
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Association Tests" href="tests.html" />
    <link rel="prev" title="Input Files" href="input.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            PangyPlot
              <img src="_static/pangyplot_logo.svg" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="local_build.html">Local Build</a></li>
<li class="toctree-l1"><a class="reference internal" href="input.html">Input Files</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Choosing Parameters</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#command-line-parameters">Command Line Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="#parameter-explaination">Parameter Explaination</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#minor-allele-frequency-cutoff">Minor Allele Frequency Cutoff</a></li>
<li class="toctree-l3"><a class="reference internal" href="#missing-data-threshold">Missing Data Threshold</a></li>
<li class="toctree-l3"><a class="reference internal" href="#filter-by-genomic-coordinate">Filter By Genomic Coordinate</a></li>
<li class="toctree-l3"><a class="reference internal" href="#must-pass">Must <em>PASS</em></a></li>
<li class="toctree-l3"><a class="reference internal" href="#read-depth-high-low-cutoff">Read Depth High/Low Cutoff</a></li>
<li class="toctree-l3"><a class="reference internal" href="#collapse-variants">Collapse Variants</a></li>
<li class="toctree-l3"><a class="reference internal" href="#testing-parameters">Testing Parameters</a></li>
<li class="toctree-l3"><a class="reference internal" href="#threads-and-batch-size">Threads and Batch Size</a></li>
<li class="toctree-l3"><a class="reference internal" href="#plot-results">Plot Results</a></li>
<li class="toctree-l3"><a class="reference internal" href="#explain-filter">Explain Filter</a></li>
<li class="toctree-l3"><a class="reference internal" href="#retain-genotypes">Retain Genotypes</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="tests.html">Association Tests</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="power_calculation.html">Power Simulation Package</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">PangyPlot</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Choosing Parameters</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/parameters.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="choosing-parameters">
<span id="parameters"></span><h1>Choosing Parameters<a class="headerlink" href="#choosing-parameters" title="Permalink to this heading"></a></h1>
<section id="command-line-parameters">
<h2>Command Line Parameters<a class="headerlink" href="#command-line-parameters" title="Permalink to this heading"></a></h2>
<p>A command line version of VikNGS is available for users who wish to do association testing without running a user interface. The command line tool requires specification of a <a class="reference internal" href="input.html#multisample-vcf"><span class="std std-ref">multi-sample VCF file</span></a> and corresponding <a class="reference internal" href="input.html#sample-info"><span class="std std-ref">sample information file</span></a>. By default, the command will run a common association tests on a single thread.</p>
<p>Run <code class="docutils literal notranslate"><span class="pre">vikNGS</span> <span class="pre">-h</span></code> for the list of relevant commands.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Parameter</p></th>
<th class="head"><p>Value/Default</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><strong>-\-vcf, -v</strong></p></td>
<td><p>[DIRECTORY]</p></td>
<td><p>Directory of a multi-sample VCF file (required)</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-sample, -i</strong></p></td>
<td><p>[DIRECTORY]</p></td>
<td><p>Directory of a file containing sample information (required)</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-bed,-b</strong></p></td>
<td><p>[DIRECTORY]</p></td>
<td><p>directory of a BED file for collapsing variants</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-out, -o</strong></p></td>
<td><p>[DIRECTORY]= .</p></td>
<td><p>Directory for output (defaults to current directory)</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-help, -h</strong></p></td>
<td></td>
<td><p>Print a help message and exit</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-common, -c</strong></p></td>
<td></td>
<td><p>Perform a common variant association test (default)</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-rare, -r</strong></p></td>
<td><p>[TEST NAME]</p></td>
<td><p>Perform a rare variant association test</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-boot, -n</strong></p></td>
<td><p>[INT]=1000</p></td>
<td><p>Number of bootstrap iterations to calculate</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-stop, -s</strong></p></td>
<td></td>
<td><p>Stop bootstrapping if p-value looks to be &gt; 0.05</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-collapse, -k</strong></p></td>
<td><p>[INT]=5</p></td>
<td><p>Collapse every k variants (rare only)</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-gene</strong></p></td>
<td></td>
<td><p>Collapse variants by gene if BED file specified (default)</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-exon</strong></p></td>
<td></td>
<td><p>Collapse variants by exon if BED file specified</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-from</strong></p></td>
<td><p>[INT]</p></td>
<td><p>Only include variants with <strong>POS</strong> larger than this value</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-to</strong></p></td>
<td><p>[INT]</p></td>
<td><p>Only include variants with <strong>POS</strong> smaller than this value</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-chr</strong></p></td>
<td><p>[CHR NAME]</p></td>
<td><p>Only include variants on this chromosome</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-maf, -m</strong></p></td>
<td><p>[FLOAT]=0.05</p></td>
<td><p>Minor allele frequency cut-off (common-rare threshold)</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-depth, -d</strong></p></td>
<td><p>[INT]=30</p></td>
<td><p>Read depth cut-off (low-high read depth threshold)</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-missing, -x</strong></p></td>
<td><p>[FLOAT]=0.1</p></td>
<td><p>Missing data cut-off (maximum tolerance for missing data)</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-all, -a</strong></p></td>
<td></td>
<td><p>Include variants which do not have <em>PASS</em> in the <strong>FILTER</strong> column</p></td>
</tr>
<tr class="row-odd"><td><p><strong>-\-threads, -t</strong></p></td>
<td><p>[INT]=1</p></td>
<td><p>Number of threads</p></td>
</tr>
<tr class="row-even"><td><p><strong>-\-batch, -h</strong></p></td>
<td><p>[INT]=1000</p></td>
<td><p>Number of variants to read from VCF before beginning tests</p></td>
</tr>
</tbody>
</table>
<p><strong>Example 1.</strong> Running a common test on 16 threads for variants on chromosome 7 with minor allele frequency &gt; 10% and ignoring what is in the <strong>FILTER</strong> column of the VCF:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">./</span><span class="n">VikNGS</span> <span class="o">--</span><span class="n">vcf</span> <span class="p">[</span><span class="o">...</span><span class="p">]</span> <span class="o">--</span><span class="n">sample</span> <span class="p">[</span><span class="o">...</span><span class="p">]</span> <span class="o">--</span><span class="nb">chr</span> <span class="n">chr7</span> <span class="o">-</span><span class="n">m</span> <span class="mf">0.1</span> <span class="o">--</span><span class="nb">all</span> <span class="o">-</span><span class="n">t</span> <span class="mi">16</span>
</pre></div>
</div>
<p><strong>Example 2.</strong> Running a rare test (CAST) on 4 threads, collapsing variants along genes and using one million bootstrap iterations with early stopping:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">./</span><span class="n">VikNGS</span> <span class="o">--</span><span class="n">vcf</span> <span class="p">[</span><span class="o">...</span><span class="p">]</span> <span class="o">--</span><span class="n">sample</span> <span class="p">[</span><span class="o">...</span><span class="p">]</span> <span class="o">--</span><span class="n">bed</span> <span class="p">[</span><span class="o">...</span><span class="p">]</span> <span class="o">-</span><span class="n">r</span> <span class="n">cast</span> <span class="o">--</span><span class="n">gene</span> <span class="o">-</span><span class="n">n</span> <span class="mi">1000000</span> <span class="o">--</span><span class="n">stop</span> <span class="o">-</span><span class="n">t</span> <span class="mi">4</span>
</pre></div>
</div>
</section>
<section id="parameter-explaination">
<span id="explain-param"></span><h2>Parameter Explaination<a class="headerlink" href="#parameter-explaination" title="Permalink to this heading"></a></h2>
<section id="minor-allele-frequency-cutoff">
<h3>Minor Allele Frequency Cutoff<a class="headerlink" href="#minor-allele-frequency-cutoff" title="Permalink to this heading"></a></h3>
<p>While reading the VCF file, VikNGS computes an allele frequency for each variant. The minor allele frequency (MAF) is estimated only using the samples included in the multisample VCF file. The MAF cutoff is used to define which variants are considered “rare” versus “common”. When running a common association test, variants with estimated minor allele frequencies <em>less than</em> the MAF cutoff (ie. rare variants) will be excluded from testing. Likewise, when running a rare association test, variants with estimated minor allele frequencies <em>greater than</em> the MAF cutoff (ie. common variants) will be excluded from testing.</p>
</section>
<section id="missing-data-threshold">
<h3>Missing Data Threshold<a class="headerlink" href="#missing-data-threshold" title="Permalink to this heading"></a></h3>
<p>Variants may have ambiguous or missing genotype information (ex. <strong>GT</strong> = ./.) for some of the individuals in the multi-sample VCF file. If too much data is missing, association tests may produce misleading results. Any variant that is missing more data than this threshold will be excluded from testing. The default value is 0.1 which means if more than 10% of sample calls cannot be determined, the variant will be ignored.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If running a quantitative association test, the proportion of missing data will be calculated from all samples. In a case-control test, two proportions will be calculated (one for all cases, one for all controls) if either cases <em>or</em> controls fail to satisfy the missing threshold, the variant will be excluded.</p>
</div>
</section>
<section id="filter-by-genomic-coordinate">
<h3>Filter By Genomic Coordinate<a class="headerlink" href="#filter-by-genomic-coordinate" title="Permalink to this heading"></a></h3>
<p>Enables filtering of variants based on the <strong>CHR</strong> and <strong>POS</strong> values in the VCF file. Variants outside a specific chromosome or range of positions can be excluded.</p>
</section>
<section id="must-pass">
<h3>Must <em>PASS</em><a class="headerlink" href="#must-pass" title="Permalink to this heading"></a></h3>
<p>Variants which do not contain “<em>PASS</em>” in the <strong>FILTER</strong> column of the VCF are filtered out. By default this filtering step is on, turning it off will cause the contents of the <strong>FILTER</strong> column to be ignored.</p>
</section>
<section id="read-depth-high-low-cutoff">
<h3>Read Depth High/Low Cutoff<a class="headerlink" href="#read-depth-high-low-cutoff" title="Permalink to this heading"></a></h3>
<p>Samples with read depth above this threshold are considered high read depth samples (default=30). This is only used for the vRVS test if read depth values are provided in the sample infomation file.</p>
</section>
<section id="collapse-variants">
<h3>Collapse Variants<a class="headerlink" href="#collapse-variants" title="Permalink to this heading"></a></h3>
<p>See information on the <a class="reference internal" href="input.html#bed-file"><span class="std std-ref">BED file section</span></a> on the Input page for details.</p>
</section>
<section id="testing-parameters">
<h3>Testing Parameters<a class="headerlink" href="#testing-parameters" title="Permalink to this heading"></a></h3>
<p>See information on the <a class="reference internal" href="tests.html#tests"><span class="std std-ref">Tests</span></a> page for details on the tests available.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Use <code class="docutils literal notranslate"><span class="pre">-r</span> <span class="pre">cast</span></code> and <code class="docutils literal notranslate"><span class="pre">-r</span> <span class="pre">skat</span></code> for the CAST-like and SKAT-like tests, respectively.</p>
</div>
</section>
<section id="threads-and-batch-size">
<h3>Threads and Batch Size<a class="headerlink" href="#threads-and-batch-size" title="Permalink to this heading"></a></h3>
<p>Number of threads to perform association testing on. Batch size is the number of variants to process at one time on a given thread.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>VikNGS will parse the VCF file line-by-line and store the data in memory. When using a large batch size, please keep in mind the memory limits of your device as these settings will determine how much memory is used.</p>
</div>
</section>
<section id="plot-results">
<h3>Plot Results<a class="headerlink" href="#plot-results" title="Permalink to this heading"></a></h3>
<p>Only available on the graphical user interface. A plotting interface will be displayed following the association testing in a new window if this setting is checked.</p>
</section>
<section id="explain-filter">
<h3>Explain Filter<a class="headerlink" href="#explain-filter" title="Permalink to this heading"></a></h3>
<p>Writes a file that explain why filtered variants were filtered if checked. See <cite>Output &lt;output&gt;</cite> for more details.</p>
</section>
<section id="retain-genotypes">
<h3>Retain Genotypes<a class="headerlink" href="#retain-genotypes" title="Permalink to this heading"></a></h3>
<p>This setting will store genotypes parsed from the VCF file in memory and will enable exploration of these values after p-values have been calculated.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Retaining all genotypes is extremely memory-intensive since a large amount of the data from the VCF file is being stored in memory simultaneously. Please only use this option for small datasets or on machines with very large amounts of memory.</p>
</div>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="input.html" class="btn btn-neutral float-left" title="Input Files" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="tests.html" class="btn btn-neutral float-right" title="Association Tests" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Scott Mastromatteo.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>